package/network/utils/wireguard-tools/files/wireguard.init

```diff
@@ -0,0 +1,122 @@
+#!/bin/sh /etc/rc.common
+
+START=80
+USE_PROCD=1
+
+WG_DIR="/tmp/wireguard"
+
+wireguard_check_peer(){
+	local cfg="${1}"
+	local cfile="${2}"
+
+	uci show "network.${cfg}" >> "${cfile}"
+}
+
+wireguard_check_interface() {
+	local cfg="${1}"
+	local proto cfile n_sum o_sum
+
+	config_get proto "${cfg}" proto
+	[ "${proto}" = "wireguard" ] || return 0
+	cfile="$(mktemp -p "${WG_DIR}")"
+	config_foreach wireguard_check_peer "wireguard_${1}" "${cfile}"
+
+	. /lib/functions/network.sh
+
+	n_sum="$(md5sum "${cfile}" | cut -d" " -f1)"
```

Comment:
Why is there md5 here?

Reply:
I would have needed this to determine if the configuration has changed, to delete peer section when the inteface is deleted. But as @jow- already recognized this is not ok https://github.com/openwrt/openwrt/pull/2625/files/c0befebe47cd917e1b3a100dc58c9d848744083f#r378338317

Reply:
Still doesn't seem like a good reason for md5.

Reply:
While I'm aware that MD5 isn't good for various security tasks it should still be just fine for simple cases like comparing configs.

If you check `/sbin/reload_config` it's exactly how comparing config files it handled. Using `md5sum`.